src/main/java/uk/gov/digital/ho/hocs/workflow/api/WorkflowService.java

```diff
@@ -357,4 +362,45 @@ public void updateStage(UUID caseUUID, UUID stageUUID, Map<String, String> value
 
         camundaClient.completeTask(stageUUID, values);
     }
+
+    public ResponseEntity closeCase(UUID caseUUID) throws JsonProcessingException {
+        GetCaseworkCaseDataResponse caseDetails = caseworkClient.getCase(caseUUID);
+
+        //Get stage uuid from casework
+        GetStagesResponse stage = caseworkClient.getActiveStage(caseDetails.getReference().replace("/", "%2F"));
+        UUID stageUUID = stage.getStages().stream().findFirst().get().getStageUUID();
+
+        //Mark case as complete
+        caseworkClient.completeCase(caseUUID, true);
+
+        //Update the stage team to null
+        UUID oldTeam = caseworkClient.getStageTeam(caseUUID, stageUUID); //To allow reversion if required
+        try {
+            caseworkClient.updateStageTeam(caseUUID, stageUUID, null, null);
+        } catch(Exception e) { //Revert marking as complete and return error
+            caseworkClient.completeCase(caseUUID, caseDetails.getCompleted()); //Audit event may already have been created for closed case
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Failed to update stage team for " + caseUUID + " Error : " + e);
+        }
+
+        //Delete Camunda process
+        try{
+            deleteProcess(stageUUID);
+        } catch(Exception e) { //Revert team change and case complete and return error
+            caseworkClient.updateStageTeam(caseUUID, stageUUID, oldTeam, null);
+            caseworkClient.completeCase(caseUUID, caseDetails.getCompleted()); //Audit event may already have been created for closed case
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Failed to delete process for " + caseUUID + " Error : " + e);
+        }
+        return ResponseEntity.ok("Closed case " + caseUUID);
+    }
+
+    private String getStageUUID(String stage) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode node = objectMapper.readTree(stage).get("stages").get(0).get("uuid");
```

Comment:
Would mapping that into a object be simpler to read?

Reply:
This method was no longer used after getting a DTO rather than a string in an earlier review comment so I've removed it